source/tyr/tyr/tasks.py

```diff
@@ -146,28 +124,78 @@ def import_data(
         # currently the name of a dataset is the path to it
         dataset.name = filename
         models.db.session.add(dataset)
+
+        # Create a job by type of data
+        job = models.Job()
+        job.instance = instance
+        job.state = 'pending'
         job.data_sets.append(dataset)
 
+        if new_action:
+            models.db.session.add(job)
+            models.db.session.commit()
+            # We pass the job id to each tasks, but job need to be commited for having an id
+            new_action.kwargs['job_id'] = job.id
+            actions.append(new_action)
+            actions.append(finish_job.si(job.id))
+
     if actions:
+        # Create a new job and dataset for ed2nav that will generate binary file (New .nav.lz4)
+        dataset = models.DataSet()
+        dataset.type = dataset.family_type = 'nav'
+        dataset.name = instance_config.target_file
+        models.db.session.add(dataset)
+
+        job = models.Job()
+        job.instance = instance
+        job.state = 'pending'
+        job.data_sets.append(dataset)
         models.db.session.add(job)
         models.db.session.commit()
-        # We pass the job id to each tasks, but job need to be commited for having an id
-        for action in actions:
-            action.kwargs['job_id'] = job.id
-        # Create binary file (New .nav.lz4)
+
         binarisation = [ed2nav.si(instance_config, job.id, custom_output_dir)]
         actions.append(chain(*binarisation))
```

Comment:
Why make a chain of one element?

Reply:
Legacy code. I asked the question once in slack but it stayed without answer:
https://kisiodigital.slack.com/archives/GBV70C70S/p1575562963043200